### PR TITLE
Azure rspec smoke tests

### PR DESCRIPTION
**Pull request by alexsomesan**
Just porting verbatim from previous smoke.sh
Was a low hanging fruit after implementing most of the logic for Azure private.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -226,101 +226,47 @@ pipeline {
...
-                                sh """#!/bin/bash -x
-                                  ${WORKSPACE}/tests/smoke/azure/smoke.sh destroy_azure_cli vars/basic.tfvars
-                                """
-                              }
-                            }
-                          }
-                          cleanWs notFailBuild: true
-                        }
-                      }
+                      checkout scm
```
Comment (cpanato) [inline Jenkinsfile]: please use ` unstash 'repository'` instead of `checkout scm` and `unstash installer/smoke`

File: Jenkinsfile [hunk 2]
```diff
@@ -226,101 +226,47 @@ pipeline {
...
+                      unstash 'installer'
+                      unstash 'smoke'
+                      sh """#!/bin/bash -ex
+                        cd tests/rspec
+                        bundle exec rspec spec/azure_basic_spec.rb
+                      """
+                      deleteDir()
```
Comment (cpanato) [inline Jenkinsfile]: please use `cleanWs notFailBuild: true` instead of `deleteDir()`

File: Jenkinsfile (hunk 1 above)
Comment (alexsomesan) [reply Jenkinsfile]: Done

File: Jenkinsfile (hunk 2 above)
Comment (alexsomesan) [reply Jenkinsfile]: Done

**Comment by alexsomesan** [comment]
retest this please

**Comment by alexsomesan** [comment]
retest this please

**Comment by alexsomesan** [comment]
retest this please

**Comment by alexsomesan** [comment]
retest this please

**Comment by alexsomesan** [comment]
retest this please

**Comment by alexsomesan** [comment]
Bare metal is at it again. The rest of the suite (including all Azure tests) is green.

**Pull request merged**